substrate/utils/frame/remote-externalities/src/lib.rs

```diff
@@ -893,34 +893,70 @@ where
 
 		let at = self.as_online().at_expected();
 
-		let client = self.as_online().rpc_client();
+		// Create an Arc to share ownership of the client.
+		let client = Arc::new(self.as_online().rpc_client().clone());
 		let mut child_kv = vec![];
-		for prefixed_top_key in child_roots {
-			let child_keys =
-				Self::rpc_child_get_keys(client, &prefixed_top_key, StorageKey(vec![]), at).await?;
 
-			let child_kv_inner =
-				Self::rpc_child_get_storage_paged(client, &prefixed_top_key, child_keys, at)
+		// Spawn tasks for each child root and collect their handles.
```

Comment:
since this is network bound (not cpu bound), we can use use async/await on a single thread rather than spawning multiple threads. simpler, same performance, and has less issues with jsonrpsee. example: https://github.com/paritytech/polkadot-sdk/blob/afbb749b4cdc7a924da37b5f604f847a4791e8d8/substrate/utils/frame/remote-externalities/src/lib.rs#L729-L734

Reply:
Okay. Updated the code accordingly.

`wss://rococo-contracts-rpc.polkadot.io:443` will not work because the node is not configured with the `--rpc-max-request-size 10000000 --rpc-max-response-size` flags.

Ideally we can test this with chopsticks. I ran:
```
npx @acala-network/chopsticks@latest --endpoint=wss://rococo-contracts-rpc.polkadot.io:443
```
Which gave me a running cloned version of the Rococo Contracts node.

However, when I ran:
```
TEST_WS=ws://127.0.0.1:8000 cargo test --features=remote-test -p frame-remote-externalities -- --nocapture
```
I got the following error:
```
⠋     0.002 s   Loading snapshot...Feb 20 14:58:50.169 ERROR remote-ext: Error = Transport(Request rejected `426`) 
```
Thoughts?

Reply:
> wss://rococo-contracts-rpc.polkadot.io:443 will not work because the node is not configured with the --rpc-max-request-size 10000000 --rpc-max-response-size flags.

It should not need those flags. I just scraped from that rpc using latest version of try-runtime-cli locally, what error do you get?

Chopsticks will be unusable for scraping the entire state, afaik it makes a single request for each state entry..

Reply:
Hey @liamaharon, kinda stumbled here, can you describe to me the steps required to run try-runtime-cli against `rococo-contracts-rpc.polkadot.io:443`? I would use `follow-chain` with the `substrate-contracts-node`?

Reply:
Try `try-runtime create-snapshot --uri wss://rococo-contracts-rpc.polkadot.io:443 ../snaps/rococo-contracts.snap`

Reply:
Looks like the error is coming from JSONrpsee:
```rust
/// Max number of request slots exceeded.
#[error("Max concurrent requests exceeded")]
MaxSlotsExceeded,
```
https://github.com/paritytech/jsonrpsee/blob/6da4608bee6e1023257b780bfe8d1778c0fad4e3/core/src/client/error.rs#L57-L59

Reply:
Try limit the concurrency to the `PARALLEL_REQUESTS` https://github.com/paritytech/polkadot-sdk/blob/9bf1a5e23884921498b381728bfddaae93f83744/substrate/utils/frame/remote-externalities/src/lib.rs#L354

Reply:
Removed `Arc` and limited concurrency to `PARALLEL_REQUESTS`.